lib/ssl/doc/src/ssl.xml

```diff
@@ -1461,15 +1461,32 @@ fun(srp, Username :: binary(), UserState :: term()) ->
       <datatype>
 	<name name="server_session_tickets"/>
 	<desc>
-	  <p>Configures the session ticket functionality. Allowed values are <c>disabled</c>,
-	  <c>stateful</c> and <c>stateless</c>.</p>
-	  <p>If it is set to <c>stateful</c> or
-	  <c>stateless</c>, session resumption with pre-shared keys is enabled and the server will
-	  send stateful or stateless session tickets to the client after successful connections.</p>
+          <p>Configures the session ticket functionality. Allowed values are <c>disabled</c>,
+          <c>stateful</c>, <c>stateless</c>, <c>stateful_with_cert</c>, <c>stateless_with_cert</c>.</p>
+          <p>If it is not set to <c>disabled</c>,
+          session resumption with pre-shared keys is enabled and the server will
+          send stateful or stateless session tickets to the client after successful connections.</p>
+
+          <warning><p>If it is set to <c>stateful</c> or <c>stateless</c>
```

Comment:
I think we probably should make this a note instead of a warning. A warning is usually used to alert you that something is less secure, very memory-consuming, or might be a bad idea under some circumstances.  I was thinking something along the lines: Pre-shard key session ticket resumption does not include any certificate exchange, hence the function ssl:peercert /1 will not be able to return the peer certificate as it is only communicated in the initial handshake. The 
server options <c>stateful_with_cert</c> or <c>stateless_with_cert</c> may be used to make a server associate the client certificate from the original handshake with the tickets it issues. 

Here we might need a warning that using these options will increase memory consumption for stateful tickets and
for stateless tickets it will increase the handshake payload.

Reply:
Makes sense, I changed it into a note with your text, and then I changed the last paragraph into a warning specifying the increased memory consumption and increased payload size. What do you think?